### DISCUSSION TITLE
How do I specify specific cmake and powershell versions to keep abi's constant

### DISCUSSION THREAD

**Discussion by senstar-nross** [microsoft/vcpkg#27879] [Q&A]
We have setup binary caching so that everyone can benefit from prebuilt vcpkgs.  Only the build machine gets to publish the binaries to a central location that everyone references.

We also use our own triplet overlay instead of x64-windows with things like:

set(VCPKG_PLATFORM_TOOLSET v141)
set(VCPKG_DISABLE_COMPILER_TRACKING true)
set(VCPKG_VISUAL_STUDIO_PATH "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Professional")

also setting both the VCPKG_DEFAULT_HOST_TRIPLET and VCPKG_DEFAULT_TRIPLET environment variables

I have even gone so far as to fork vcpkg and all team members use the exact same version of vcpkg

all of this to avoid rebuilding for minor differences like visual studio minor revisions etc... 

But recently I have noticed a number of people are complaining about everything rebuilding.  The problem is that a number of people seem to be getting different versions of cmake and powershell.  

I have tried VCPKG_FORCE_DOWNLOADED_BINARIES  but we still seem to have different cmake and powershell versions.
Do I have to also download and install specific versions of powershell and cmake and update the path and use FORCE_SYSTEM_BINARIES?  I really thought VCPKG_FORCE_DOWNLOADED_BINARIES would be the ticket.  

Any suggestions on how to do this?  Every time this happens people start to suffer through 2+ hour builds of opencv, boost etc...
How are powershell and cmake getting updated.. One person suggested that their powershell was updated by windows update.  How does vcpkg decide what versions to download when using  VCPKG_FORCE_DOWNLOADED_BINARIES.

**Comment by autoantwort** [comment]
Use `--x-abi-tools-use-exact-versions`
